source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -145,6 +147,10 @@ namespace aspect
         //
         // The rotation matrix is a direction cosine matrix, representing the orientation of the grain in the domain.
         // The fabric is determined later in the computations, so initialize it to -1.
+
+#ifndef ASPECT_WITH_WORLD_BUILDER
+        (void)position;
```

Comment:
Could you move this down into the `else` branch of the `ifdef` below? I.e. around line 191 or so? Then you dont need another ifdef here and the code is easier to understand.